examples/hmc.py

```diff
@@ -272,8 +280,20 @@ def stats(self):
 def main(**kwargs):
     args = Namespace(**kwargs)
 
-    lattice = (args.lattice_size,) * 4
-    helper = HMCHelper(args.beta, lattice)
+    glattice = (args.lattice_size,) * 4
+    llattice = ()
+    procs = args.procs
+    for ldim, cdim in zip(glattice, procs):
+        if not (ldim / cdim).is_integer():
+            raise ValueError(
+                "Each lattice dim needs to be divisible by the corresponding dim of the Cartesian communicator!"
+            )
+        llattice += (int(ldim / cdim),)
+    comm = get_cart(procs)
+    lib.set_comm(comm)
+    lib.init_quda()
```

Comment:
would be nice if all of the above becomes part of lyncs_quda. e.g.
```
glattice = (args.lattice_size,) * 4
lib.set_procs(args.procs)
llattice = lib.local_lattice(glattice)
```
and I would avoid initializing and finalizing `lib` explicitly if possible